### DISCUSSION TITLE
How to turn off app verification?

### DISCUSSION THREAD

**Discussion by mgubaidullin** [jbangdev/jbang#1677] [Q&A] [answered]
I have installed Camel app into jbang in Docker container:
```
jbang trust add  --quiet  https://github.com/apache/camel \
&& jbang app install camel@apache/camel
```

However jbang always check if [CamelJBang.java](https://github.com/apache/camel/blob/HEAD/dsl/camel-jbang/camel-jbang-main/dist/CamelJBang.java) has changed and download CamelJbang.java if it changed.
Is it possible to turn off this verification and always use Camel installed?

**Comment by quintesse** [comment, marked as answer]
Try `jbang config set cache-evict never`, that will force jbang to always use any cached files instead of downloading them.

Btw: this will set this behaviour globally, there's no easy way to do this only for that command. (If you want I can explain how to do so, but it's somewhat of a hack)

**Comment by maxandersen** [comment]
@mgubaidullin jbang --offline should work too!

**Reply by maxandersen** [reply]
assuming you already ran it once.

if you want the command to run no matter if internet access then yes @quintesse cache evict will be needed..but yeah; thats very global.

Whats your usecase?

**Reply by mgubaidullin** [reply]
afaik --offline won't download dependencies. I just need to set camel-jbang in cache in docker image

**Reply by mgubaidullin** [reply]
I have a Docker image with Java, Maven, Git, Jbang and Camel-Jbang 
```
jbang trust add  --quiet  https://github.com/apache/camel \
    && jbang app install camel@apache/camel \
    && jbang config set cache-evict never
```

https://github.com/apache/camel-karavan/blob/main/karavan-web/docker/Dockerfile.devmode

When `jbang ... camel@apache/camel` executes jbang verify camel-jbang, if changed in github jbang download it and replace to the latest version.

I need to fix camel-jbang to the version installed during docker image build.

Looks like `jbang config set cache-evict never` responds to my demand, isn't it?

**Reply by maxandersen** [reply]
If you already built it (install should check that already) in a previous step in your docker file then you already have the dependencies.

Therefore jbang --offline <script> should be sufficient. 

If not - I consider that a bug. 

Cache evict setting should work too but only for the script.
